### ISSUE TITLE
Model 1 results for warping is unexpected

### ISSUE THREAD

**Issue by pratiklodha95** [shionhonda/viton-gan#4]
I implemented the code as mention in this repo and cp-vton repo.

Everything is working fine, but the model results from the part 1 model is not as expected,
Would like to if there is a specific formats of input one has to give, I have kept all inputs as per your code and data preparation code from cp-vton repo.

![image](https://user-images.githubusercontent.com/6401006/66098849-eaaa7800-e5c1-11e9-893b-682830b01881.png)

for simple inputs like
image
![image](https://user-images.githubusercontent.com/6401006/66098858-f1d18600-e5c1-11e9-8521-96b46f942af9.png)

and
image
![image](https://user-images.githubusercontent.com/6401006/66098862-f7c76700-e5c1-11e9-8393-23de13cb3f7f.png)

**Comment by shionhonda** [comment]
Could you give me more information?

- How did you obtain `warp-person.jpg`? Is it a warped cloth layered on the person image?
- What did you input? You need cloth mask, parsed person (by LIP) and pose (by OpenPose) in addition to the pair of person and cloth.

Also, I'm not sure about the extrapolation performance of VITON-GAN. The dataset I used only contains female models.

**Comment by jakubLangr** [comment]
Hi @shionhonda @pratiklodha95 I got the same results. Even on the original test files, using the provided GMM epoch 99 file.

![000038_1](https://user-images.githubusercontent.com/1884004/66223642-dce11980-e6cb-11e9-8c91-f08d40df19ef.jpg)

Thoughts?

**Comment by shionhonda** [comment]
Thanks for reporting @jakubLangr .  
That is weird because the result in my local repository looks fine.
![000038_1](https://user-images.githubusercontent.com/17271286/66278964-f57a4b00-e8e8-11e9-8ebc-2b92479f1dd4.jpg)

I might have provided different GMM model file. Let me check.

**Comment by shionhonda** [comment]
I did reproduce the successful result by 1) cloning this repository, 2) downloading the dataset and the trained model `gmm_epoch_99.pth`, and 3) running `python run_gmm.py`.  
Could you check if the trained model is successfully loaded, please? I'm wondering if TPS transform is really working in your case.

**Comment by jakubLangr** [comment]
Hi @shionhonda I went ahead and trained your algorithm from scratch. These are the results from my training run over the weekend. The images seem within norms so it seems that the training code for GMM is working on my machine.
![image](https://user-images.githubusercontent.com/1884004/66304683-b272a480-e8f5-11e9-951d-dba50c47c763.png)
![image](https://user-images.githubusercontent.com/1884004/66304691-b7375880-e8f5-11e9-88d1-86bc42a6a5c2.png)

However, I am fairly confident that the pretrained `epoch_99` checkpoint is used correctly, as I do not get any file not found errors or similar and the path seems to match the requirement.

I do agree with your suggestion that it is likely that TPS is not working, but I am not sure why.

I am now re-running it on the freshly re-trained model.

Out of curiosity, what setup are you using (Pytorch / Cuda versions?) & OS?

**Comment by jakubLangr** [comment]
On re-run with my re-trained model the results are somewhat better. But I am still confused as to why the neck gets removed?

![000010_1](https://user-images.githubusercontent.com/1884004/66314355-d5a84e80-e90b-11e9-8d6b-88474af550d4.jpg)

**Comment by shionhonda** [comment]
@jakubLangr 
Glad to see that your re-running looks fine. But I'm not sure why as well.
I used the setup like:

- PyTorch 1.2.0 (it should be fine if > 1.0)
- CUDA 9.0
- Ubuntu 16.04

Necks are removed in LIP.
http://sysu-hcp.net/lip/parsingchallenge.php
